src/cipher.rs

```diff
@@ -0,0 +1,71 @@
+use aes::Aes128;
+use cipher::KeyInit;
+use scrypt_jane::scrypt::{scrypt, ScryptParams};
+
+#[derive(Debug)]
+pub(crate) struct AesCipher {
+    pub(crate) aes: Aes128,
+    pub(crate) nonce: u32,
+}
+
+impl AesCipher {
+    pub(crate) fn new(challenge: &[u8; 32], nonce: u32, params: ScryptParams) -> Self {
+        let mut key = [0u8; 16];
+        scrypt(challenge, &nonce.to_le_bytes(), params, &mut key);
```

Comment:
I think that scrypt was chosen because of its ASIC-resistance. @fasmat could you help answering?

Reply:
The reason why we hash the challenge and the nonce is because we want checking part of the post data with many nonces to be more costly than checking the full data with few nonces.

Scrypt serves better as a proof of work in this regard and was suggested by the research team.

As to why we are using scrypt + keccak instead of scrypt + blake3: because we are already using the former for label creation and won't change that before genesis.

If at any point we switch to scrypt+blake for label generation we can do that here as well.